frontend/app/ui/subscriptions/list/controller.js

```diff
@@ -11,21 +12,52 @@ export default class SubscriptionsListController extends Controller {
   @service intl;
 
   @tracked projects;
+  @tracked searches = {};
+  @tracked sortBy = {};
 
   breadcrumbs = [
     { label: this.intl.t("page.subscriptions.title"), route: "subscriptions" },
     { label: this.intl.t("page.subscriptions.list.title") },
   ];
 
-  @action search(key, query) {
-    const normalized = query.trim().toLowerCase();
+  @action searchFor(key, query) {
+    if (isEmpty(query)) {
+      delete this.searches[key];
+    } else {
+      this.searches[key] = query;
+    }
+
+    this.search();
+  }
+
+  @action limitCritical(event) {
+    const { checked } = event.target;
+
+    if (checked) {
+      this.searches.isTimeAlmostConsumed = "true";
+    } else {
+      delete this.searches.isTimeAlmostConsumed;
+    }
+
+    this.search();
+  }
 
-    this.projects =
-      normalized.length > 0
-        ? this._projects.filter((project) => {
-            return project.get(key).toLowerCase().includes(normalized);
-          })
-        : this._projects;
+  @action search() {
+    let projects = this._projects;
+
+    for (const key in this.searches) {
+      const normalized = String(this.searches[key]).trim().toLowerCase();
+
+      if (isEmpty(normalized)) {
+        continue;
+      }
+
+      projects = projects.filter((project) => {
+        return String(project.get(key)).toLowerCase().includes(normalized);
+      });
```

Comment:
If the above is not an option.
```suggestion
      projects = projects.filter((project) =>
        String(project.get(key)).toLowerCase().includes(normalized);
      );
```

Reply:
> If the above is not an option.

What do you mean by that?